### DISCUSSION TITLE
Is there a option for `family` with code generated `NotifierProvider`?

### DISCUSSION THREAD

**Discussion by nilsreichardt** [rrousselGit/riverpod#2628] [Q&A] [answered]
I searching for an option to have code generated `NotifierProvider` with the `family` option. My current provider looks like:

```dart
final cardFeedbackStatusProvider =
    StateProvider.autoDispose.family<CardFeedbackStatus, CardId>((ref, cardId) {
      ref.watch(cardFeedStatusControllerProvider);
  return CardFeedbackStatus.notReviewed;
});
```

Now I want to migrate this provider to the code generated `NotifierProvider`. But I couldn't find any information about the `family` option for this provider.

**Comment by AhmedLSayed9** [comment, marked as answer]
You can pass parameters directly through the `build` method:
```dart
@riverpod
class Some extends _$Some {
  @override
  SomeValue build(String param1, int param2) {
    ..
  }
}
```
It can be named parametes, optional or default values.

**Reply by nilsreichardt** [reply]
Ah, nice! I didn't know that you can pass parameters to the `build` method. Thanks!
